### PR TITLE
Expand merge

### PR DESCRIPTION
**Pull request by thomased**
Rejigged `merge.rspec()` so it can accept an arbitrary number of `rspec` objects. Added some tests to match, and fixed up the docs. It's still a little clunky, but it's necessary (I think...) to avoid a breaking changes. Is this otherwise a bad idea for reasons I haven't thought of?

### CODE REVIEW

File: R/merge.rspec.R [hunk 1]
```diff
@@ -1,51 +1,82 @@
...
 
-merge.rspec <- function(x, y, ...) {
-
-  if (!all(is.rspec(x), is.rspec(y))) {
-    stop("One or more invalid rspec objects")
-  }
-
-  if (!all("wl" %in% names(x), "wl" %in% names(y))) {
-    stop("Cannot find valid 'wl' column in one or both input objects")
+merge.rspec <- function(x, ...) {
```
Comment (Bisaloo) [inline R/merge.rspec.R:36]: Even if we move forward with this, it's still necessary to include the `y` argument to match arguments in the generic.

File: R/merge.rspec.R [hunk 2]
```diff
@@ -1,51 +1,82 @@
...
+  # If there's only one rspec argument, handle it as before
+  if(length(rspec_args) == 1) {
+    y <- rspec_args[[1]]
+    
+    if (!all(is.rspec(x), is.rspec(y))) {
+      stop("One or more invalid rspec objects")
+    }
+    
+    if (!all("wl" %in% names(x), "wl" %in% names(y))) {
+      stop("Cannot find valid 'wl' column in one or both input objects")
```
Comment (Bisaloo) [inline R/merge.rspec.R:52]:
```suggestion
      stop("Cannot find valid 'wl' column in at least one input objects")
```

File: tests/testthat/test-S3rspec.R [hunk 3]
```diff
@@ -137,10 +137,12 @@ test_that("merge.rspec", {
   expect_s3_class(sicalis_merged, "rspec")
   expect_mapequal(sicalis_TC, sicalis_merged)
 
-  expect_error(
-    merge(sicalis_T, as.data.frame(sicalis_C)),
-    "invalid rspec objects"
-  )
+  # This now works as it should (by combining both into a single rspec object) 
+  # with the rewritten function. It should, shouldn't it?
```
Comment (Bisaloo) [inline tests/testthat/test-S3rspec.R:141]: Yes, it should run without error but I think it'll give an unexpected output. From what I understand, the second argument will be silently ignored.
This example is a good demonstration of why this change might bring more trouble than it solves.

**Review by Bisaloo — COMMENTED** [review]
I'm on the fence here. I understand where you're coming from but by doing so, we're breaking away from the standard data.frame behaviour, which:

- may be confusing
- may cause problems. I'm not completely sure if `merge.data.frame()` only works on two objects for a good technical reason or if it's an historical artifact

What about documenting that users can do `do.call(merge.rspec, list(rspec1, rspec2, rspec3))` if they want to merge more than 2 rspec objects? This is what you end up doing under the hood in this PR anyways.

**Comment by thomased** [comment]
Yeah on reflection I think you’re quite right. I threw this together quickly while playing with something else, but not sure it’s worth the risk. 

The two df limit on `merge.data.frame` is curious though, I’d be interested to know more sometime. 

Leaning toward just a richer example like you suggest then, but I’ll leave this here for a bit to think it over/chat.

**Comment by thomased** [comment]
Added the example - 9509cfbe2148f77d61a4233cff0d44caca15f0ae. Might leave it there for now then.
